Src/Plugins/Security/SEC101_054.AlibabaAccessKeyValidator.cs

```diff
@@ -0,0 +1,119 @@
+
+using Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security.Utilities;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
+{
+    public class AlibabaAccessKeyValidator : ValidatorBase
+    {
+        internal static AlibabaAccessKeyValidator Instance;
+
+        protected const string KeyNotFound = "InvalidAccessKeyId.NotFound";
+        protected const string InvalidSecret = "SDK.InvalidAccessKeySecret";
+
+        static AlibabaAccessKeyValidator()
+        {
+            Instance = new AlibabaAccessKeyValidator();
+        }
+
+        public static string IsValidStatic(ref string matchedPattern,
+                                           ref Dictionary<string, string> groups,
+                                           ref string failureLevel,
+                                           ref string fingerprint,
+                                           ref string message)
+        {
+            return IsValidStatic(Instance,
+                                 ref matchedPattern,
+                                 ref groups,
+                                 ref failureLevel,
+                                 ref fingerprint,
+                                 ref message);
+        }
+
+        public static string IsValidDynamic(ref string fingerprint, ref string message)
+        {
+            return IsValidDynamic(Instance,
+                                   ref fingerprint,
+                                   ref message);
+        }
+
+        protected override string IsValidStaticHelper(ref string matchedPattern,
+                                                      ref Dictionary<string, string> groups,
+                                                      ref string failureLevel,
+                                                      ref string fingerprintText,
+                                                      ref string message)
+        {
+            if (!groups.TryGetNonEmptyValue("account", out string account) ||
+                !groups.TryGetNonEmptyValue("password", out string password))
+            {
+                return nameof(ValidationState.NoMatch);
+            }
+
+            fingerprintText = new Fingerprint()
+            {
+                Account = account,
+                Password = password,
+            }.ToString();
+
+            return nameof(ValidationState.AuthorizedWarning);
+        }
+
+        protected override string IsValidDynamicHelper(ref string fingerprintText,
+                                                       ref string message)
+        {
+            Fingerprint fingerprint = new Fingerprint(fingerprintText);
+            string account = fingerprint.Account;
+            string password = fingerprint.Password;
+
+            try
+            {
+                // Taken from https://www.alibabacloud.com/help/doc-detail/63638.htm
+                IClientProfile clientProfile = DefaultProfile.GetProfile("cn-shanghai", account, password);
```

Comment:
>IClientProfile [](start = 16, length = 14)

can u check if any of those are idisposable?

Reply:
None are.

---
In reply to: [591817886](https://github.com/microsoft/sarif-pattern-matcher/pull/285#discussion_r591817886) [](ancestors = 591817886)